packages/backend/src/api/middleware/metrics.ts

```diff
@@ -0,0 +1,45 @@
+import { Context, Middleware, Next } from 'koa'
+
+import { Metrics } from '../../Metrics'
+
+export function createApiMetrics(metrics: Metrics): Middleware {
+  const labels = ['path', 'method', 'status_code']
+  const apiHistogam = metrics.createHistogram({
+    name: 'api_request_duration_seconds_sum',
+    help:
+      'duration histogram of  api http responses labeled with: ' +
+      labels.join(', '),
+    labelNames: labels,
+  })
+
+  return async function (ctx: Context, next: Next): Promise<void> {
+    const key = Symbol.for('request-received.startTime')
+    // eslint-disable-next-line
+    const start: number = ctx[key as any]?.getTime?.() ?? Date.now()
+
+    await next()
+
+    const { res } = ctx
+
+    const done = () => {
+      res.removeListener('finish', done)
+      res.removeListener('close', done)
+
+      const timeMs = Date.now() - start
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const path = (ctx._matchedRoute as string | undefined) ?? ctx.path
```

Comment:
matched route would include parameter ranges like `user/:id` right? i think this could be explained in a comment